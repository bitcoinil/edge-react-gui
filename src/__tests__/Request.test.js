/* globals describe it expect */
/* eslint-disable flowtype/require-valid-file-annotation */

import * as React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'

import { RequestComponent as Request } from '../components/scenes/RequestScene.js'
import { getTheme } from '../components/services/ThemeContext.js'

describe('Request', () => {
  it('should render with loading props', () => {
    const renderer = new ShallowRenderer()

    const props = {
      currencyCode: null,
      currentScene: 'request',
      edgeWallet: null,
      exchangeSecondaryToPrimaryRatio: null,
      guiWallet: null,
      loading: true,
      primaryCurrencyInfo: null,
      receiveAddress: null,
      secondaryCurrencyInfo: null,
      showToWalletModal: null,
      useLegacyAddress: null,
      wallets: {},
      theme: getTheme()
    }
    const actual = renderer.render(<Request {...props} />)

    expect(actual).toMatchSnapshot()
  })

  it('should render with loaded props', () => {
    const renderer = new ShallowRenderer()

    const props = {
      currencyCode: 'BTC',
      edgeWallet: {},
      exchangeSecondaryToPrimaryRatio: {},
      guiWallet: {},
      loading: false,
      primaryCurrencyInfo: {},
      receiveAddress: {},
      secondaryCurrencyInfo: {},
      showToWalletModal: false,
      useLegacyAddress: false,
      currentScene: 'request',
      wallets: {},
      balance: '0',
      theme: getTheme()
    }
    const actual = renderer.render(<Request {...props} />)

    expect(actual).toMatchSnapshot()
  })
})
