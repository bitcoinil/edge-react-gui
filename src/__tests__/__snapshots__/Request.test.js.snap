// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request should render with loaded props 1`] = `
<SceneWrapper
  background="header"
  hasTabs={false}
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "justifyContent": "flex-start",
        "marginHorizontal": 22,
      }
    }
  >
    <WithTheme(EdgeTextComponent)
      style={
        Object {
          "fontFamily": "Quicksand-Medium",
          "fontSize": 45,
        }
      }
    >
      Request
    </WithTheme(EdgeTextComponent)>
    <WithTheme(EdgeTextComponent)
      style={
        Object {
          "fontSize": 28,
          "marginBottom": 34,
        }
      }
    >
      You have 0 undefined
    </WithTheme(EdgeTextComponent)>
    <ExchangedFlipInput
      color="#FFFFFF"
      exchangeSecondaryToPrimaryRatio={Object {}}
      forceUpdateGuiCounter={0}
      headerText="Receive to undefined"
      inputAccessoryViewID=""
      isEditable={true}
      isFiatOnTop={true}
      isFocus={false}
      keyboardVisible={false}
      onExchangeAmountChanged={[Function]}
      onNext={[Function]}
      overridePrimaryExchangeAmount=""
      primaryCurrencyInfo={Object {}}
      secondaryCurrencyInfo={Object {}}
      topReturnKeyType="done"
    />
    <InputAccessoryView
      backgroundColor="rgba(255, 255, 255, 0.5)"
      nativeID="cancelHeaderId"
    >
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <ForwardRef
          onPress={[Function]}
          style={
            Object {
              "paddingHorizontal": 21.068661971830984,
              "paddingVertical": 9.832042253521127,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#0073D9",
                "fontSize": 22.473239436619718,
              }
            }
          >
            
          </Text>
        </ForwardRef>
        <ForwardRef
          onPress={[Function]}
          style={
            Object {
              "paddingHorizontal": 21.068661971830984,
              "paddingVertical": 9.832042253521127,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#0073D9",
                "fontSize": 22.473239436619718,
              }
            }
          >
            Done
          </Text>
        </ForwardRef>
      </View>
    </InputAccessoryView>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(255, 255, 255, .1)",
          "borderBottomWidth": 1,
          "height": 1,
          "marginTop": 17,
          "width": "110%",
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 11,
          "marginLeft": 45,
          "marginTop": 34,
          "padding": 11,
        }
      }
    >
      <QrCode
        data=""
        size={225}
      />
    </View>
    <RightChevronButton
      marginRem={
        Array [
          1,
          0,
          0.5,
          0,
        ]
      }
      onPress={[Function]}
      text="Your Receiving Wallet Address"
    />
    <WithTheme(EdgeTextComponent) />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <ShareButtons
    copyToClipboard={[Function]}
    fioAddressModal={[Function]}
    shareViaShare={[Function]}
  />
</SceneWrapper>
`;

exports[`Request should render with loading props 1`] = `
<ActivityIndicator
  animating={true}
  color="#87939E"
  hidesWhenStopped={true}
  size="large"
  style={
    Object {
      "alignSelf": "center",
      "flex": 1,
    }
  }
/>
`;
